packages/worker/src/activity.ts

```diff
@@ -186,8 +188,17 @@ export class Activity {
         } else if (this.cancelReason) {
           // Either a CancelledFailure that we threw or AbortError from AbortController
           if (err instanceof CancelledFailure) {
-            // If cancel due to activity pause, emit an application failure for the pause.
-            if (this.context.cancellationDetails?.paused) {
+            // If cancel due to activity pause or reset, emit an application failure.
+            if (this.context.cancellationDetails?.reset) {
```

Comment:
Are we confident on if clause ordering? reset vs pause?

Reply:
Between reset and pause yes. Added a conditional for explicit cancellations, ordered _before_ the `reset` branch.